### ISSUE TITLE
Consider support Hermes without using proxy

### ISSUE THREAD

**Issue by OtacilioN** [Spyna/react-store#23]
Hey Spyna, 

Thanks for that library, I use it in a React Native Project and since now it's working very well, but recently React Native made a migration from JSC to Hermes, which does not support "proxy".

Do you think that is possible to support hermes without using proxy?

**Comment by Spyna** [comment]
Hi, 
did you look for a proxy **polyfill**? It may solve the problem. 
Maybe this can help you: https://github.com/GoogleChrome/proxy-polyfill

If not, I could add an option to not use the `Proxy`, or test if the `Proxy` is available. 
Please let me know

**Comment by OtacilioN** [comment]
Hey @Spyna awesome ✨ 

I think the second option is the best one. I will make a pull request adding this. What I need to do is just add a try catch block wrapping the Proxy call, right?

**Comment by Spyna** [comment]
I was thinking about something like this: 

```JavaScript
function proxyIsSupported() {
  const scope = ('undefined' !== typeof process &&
    '[object process]' === {}.toString.call(process)) ||
  ('undefined' !== typeof navigator && navigator.product === 'Hermes' * )
    ? global
    : self
return scope['Proxy'] !== undefined
}

if(proxyIsSupported()){
store = new Proxy(...)
}

('undefined' !== typeof process &&
    '[object process]' === {}.toString.call(process)) ||
  ('undefined' !== typeof navigator && navigator.product === 'ReactNative')
    ? global
    : self
```

`*` Not sure of this part

Maybe you can create a new file that export this function, to keep the `createStore.js` clean.

**Comment by Spyna** [comment]
Please add tests in the PR

**Comment by OtacilioN** [comment]
Hey @Spyna I made some tests and create the Pull Request #24

But I don't know exactly how to make the tests, do you recommend something?

**Comment by Spyna** [comment]
Resolved in PR #25 where I added the option `proxyStore` (boolean) to avoid the usage of a `Proxy` in those environments who don't support Proxies.

I decided to use this method instead of checking if the running engine is Hermes for two reason:
* To avoid binding the library with some external frameworks/lib
* To make the code testable, mostly in CI environments where is difficult to set up a different JS engine

You can upgrade to version `1.4.0` (PR #26) to use this feature.

```JS
Usage
const options = {
  proxyStore : false
}
createStore(Component, initialValues, options);
```

@OtacilioN please let me know if this solution works for you
